pilot/pkg/model/push_context.go

```diff
@@ -524,6 +524,18 @@ func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *Config {
 		}
 	}
 
+	// when proxy config namespace equals root namespace
+	// search for the namespace local rules at last
+	if proxy.ConfigNamespace == ps.Env.Mesh.RootNamespace {
```

Comment:
So this searches for rules marked with `exportTo: '.'` in the root namespace. I am not sure we want this? @rshriram would we expect these rules to be globally applicable in this case?

Reply:
So with this change, we need to remove the following comment earlier in the code:
// NOTE: This does mean that we are effectively ignoring private dest rules in the config root namespace

Also is searching the public rules to be done before this (as is the case in the current PR) or after?

Reply:
It should be after.